src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -83,6 +83,30 @@ object Instance {
   def instancesById(instances: Seq[Instance]): Map[Instance.Id, Instance] =
     instances.map(instance => instance.instanceId -> instance)(collection.breakOut)
 
+  object Running {
+    def unapply(instance: Instance): Option[Tuple3[Instance.Id, Instance.AgentInfo, Map[Task.Id, Task]]] = instance match {
+      case Instance(instanceId, Some(agentInfo), InstanceState(Condition.Running, _, _, _), tasksMap, _, _, _) =>
+        Some((instanceId, agentInfo, tasksMap))
+      case _ =>
+        Option.empty[Tuple3[Instance.Id, Instance.AgentInfo, Map[Task.Id, Task]]]
+    }
+  }
+
+  object Scheduled {
+
+    /**
+      * Factory method for an instance in a [[Condition.Scheduled]] state.
+      *
+      * @param runSpec The run spec the instance will be started for.
+      * @param instanceId The id of the new instance.
+      * @return An instance in the scheduled state.
+      */
+    def apply(runSpec: RunSpec, instanceId: Instance.Id): Instance = {
+      val state = InstanceState(Condition.Scheduled, Timestamp.now(), None, None)
+      Instance(instanceId, None, state, Map.empty, runSpec.version, runSpec.unreachableStrategy, None)
+    }
+  }
+
```

Comment:
should we add the new conditions to the conditionHierarchy property on line 125 here?

Reply:
or maybe we don't because the task can never have this condition?

Reply:
The hierarchy is used to infer an instance condition from task conditions. However, for `Scheduled` and `Provisioned` the instance condition is set explicitly so we do not need to infer it. In the future we want to distinguish between a task status or condition and an instance goal.